pkg/apis/autoscaling/types.go

```diff
@@ -77,10 +79,90 @@ type HorizontalPodAutoscalerSpec struct {
 	MinReplicas *int32
 	// upper limit for the number of pods that can be set by the autoscaler. It cannot be smaller than MinReplicas.
 	MaxReplicas int32
-	// target average CPU utilization (represented as a percentage of requested CPU) over all the pods;
-	// if not specified the default autoscaling policy will be used.
+	// the metrics to use to calculate the desired replica count (the
+	// maximum replica count across all metrics will be used).	The
+	// desired replica count is calculated multiplying the ratio between
+	// the target value and the current value by the current number of
+	// pods.  Ergo, metrics used must decrease as the pod count is
+	// increased, and vice-versa.  See the individual metric source
+	// types for more information about how each type of metric
+	// must respond.
 	// +optional
-	TargetCPUUtilizationPercentage *int32
+	Metrics []MetricSpec
+}
+
+// a type of metric source
+type MetricSourceType string
+
+var (
+	// a metric describing a kubernetes object (for example, hits-per-second on an Ingress object)
+	ObjectSourceType MetricSourceType = "Object"
```

Comment:
Actually, I see that it has a reference in it.  Hrm.  Then use `ObjectMetricSourceType`